### PR TITLE
[ofono] call SetDefaultDataSim on Default Data Sim change

### PR DESCRIPTION
**Pull request by natur-produkt**
If Default Data Sim is changed,
then we need to call SetDefaultDataSim from
ofono service to enable updated RIL

### CODE REVIEW

File: src/indicator/nmofono/manager-impl.cpp
Comment (mardy) [inline src/indicator/nmofono/manager-impl.cpp]: `assert(bus.isConnected());`

File: src/indicator/nmofono/manager-impl.cpp
Comment (mardy) [inline src/indicator/nmofono/manager-impl.cpp]: We don't use spaces around the `if` expressions. But in this case I think that this check is not needed: even if the interface is not valid, QtDBus will just return an error (which we are ignoring).

File: src/indicator/nmofono/manager-impl.cpp
Comment (mardy) [inline src/indicator/nmofono/manager-impl.cpp]: There's no point in making an async call if you are waiting for its result anyway.

**Review by mardy — CHANGES_REQUESTED** [review]
Hi, and thanks for this!
  But I don't think that this is the right place for this change: this class does not deal directly with the Ofono D-Bus interface, but rather delegates the work to the classes under the `wwan/` subdirectory. I don't know this project well, but my guess is that you should add a method to set the default sim to the `wwan::SimManager` class and call it from here.

**Comment by natur-produkt** [comment]
Thank you very much for your review - I really appreciate it. I'm new in project, so any feedback is needed. I changed the logic a little bit, added signal to manager-impl.cpp, added slot to sim-manager.cpp and connected those two.

I know that it can be not a final version, but waiting for your review!

File: src/indicator/nmofono/manager-impl.cpp
Comment (natur-produkt) [reply src/indicator/nmofono/manager-impl.cpp]: true - changed in next push

File: src/indicator/nmofono/manager-impl.cpp
Comment (natur-produkt) [reply src/indicator/nmofono/manager-impl.cpp]: true, I changed this

**Review by mariogrip — APPROVED** [review]
LGTM :+1:

**Comment by mariogrip** [comment]
> Thank you very much for your review - I really appreciate it. I'm new in project, so any feedback is needed. I changed the logic a little bit, added signal to manager-impl.cpp, added slot to sim-manager.cpp and connected those two.
> 
> I know that it can be not a final version, but waiting for your review!

This is fine for now, but it will need to be cleaned a bit up before merging into the "xenial" branch, but for now this works great.

**Pull request merged**
